src/components.jl

```diff
@@ -50,7 +51,9 @@ function add_component!(
 
     !skip_validation && check_component(components, component)
 
-    if !deserialization_in_progress && has_time_series(component)
+    if !deserialization_in_progress &&
```

Comment:
can we make an intermediate var here to make it easier to read.

Reply:
Can you be more specific? I'm not following.

Reply:
```suggestion
    time_series_condition = !allow_existing_time_series && has_time_series(component)
    if !deserialization_in_progress && time_series_condition
```

Reply:
Its a minor thing just to understand what the condition is about

Reply:
I took the point that it wasn't easy to read and simplified the function.